### PR TITLE
Update preprocessor `#else` and `#endif` comment

### PR DESCRIPTION
**Pull request by AlexGuteniev**
More towards #351. Product files (non-test).

Skipped in the previous #3900
 * `&&` / `||`, de-morganed
 * lone `#endif`, not specifically targeted in the previous PR
 * `yals_core.h` / `yvals.h` macro introduction
 * `#elif`-containing blocks
 * Missed for some other reasons

### CODE REVIEW

File: stl/inc/__msvc_sanitizer_annotate_container.hpp
Comment (SuperWig) [inline stl/inc/__msvc_sanitizer_annotate_container.hpp:62]:
```suggestion
#endif // !defined(_INSERT_VECTOR_ANNOTATION)
```
same below.

**Comment by StephanTLavavej** [comment]
I've pushed a merge with `main` to resolve a simple conflict with #4000, then a whole bunch of changes for issues identified during code review (a few were pre-existing cleanups that fit with this PR):

* Add missing `defined`.
* Restore lost underscores.
* Add missing arrow, replace complex De Morgan with matching comment.
* Fix whitespace.
* Use proper arrow comments.
* Simply drop a horribly long comment, using our convention that we can omit them in single-line, non-nested contexts.
* Too many arrows.
* Use systematic `_HAS_CXX17` comments.
* Use systematic `_HAS_CXX20` comments.
* Use matching arrow comments.
* Simplify calling convention comments, omit pre-existing newlines.
* Fix comments around extended aligned storage.
* Drop spurious `!`.
* Add missing `!`.
* Fix damaged `!!`, use matching "floating-point exceptions disabled (default)" comment.
* Fix FBITS comments.

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
These comment changes aren't risky; we can merge them without a second maintainer approval.

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks again for improving codebase consistency! :scroll: :page_facing_up: :sparkles:

**Pull request merged**
